src/ai/backend/manager/registry.py

```diff
@@ -1381,9 +1435,11 @@ async def start_session(
                 image_infos[str(image_ref)] = resolved_image_info
                 if not resolved_image_info.image_ref.is_local:
                     is_local_image = False
-                    registry_url, registry_creds = await get_registry_info(
-                        self.shared_config.etcd, image_ref.registry
+
+                    registry_url, registry_creds = await get_container_registry_info(
+                        self.db, cast(GUID, resolved_image_info.registry_id)
```

Comment:
```suggestion
                    registry_url, registry_creds = await get_container_registry_info(
                        session, cast(GUID, resolved_image_info.registry_id)
```
readonly session is already opened in this context. Opening a new DB session is redundant and also could raise an error.
Let's pass the opened DB session to `get_container_registry_info()` and update code inside the function.